cpp/src/slice2matlab/Main.cpp

```diff
@@ -719,333 +719,33 @@ namespace
-        //
-        string::size_type pos = 0;
-        do
-        {
-            pos = comment.find('<', pos);
-            if (pos != string::npos)
-            {
-                string::size_type endpos = comment.find('>', pos);
-                if (endpos == string::npos)
-                {
-                    break;
-                }
-                comment.erase(pos, endpos - pos + 1);
-            }
-        } while (pos != string::npos);
-
-        const string link = "{@link";
-        pos = 0;
-        do
-        {
-            pos = comment.find(link, pos);
-            if (pos != string::npos)
-            {
-                comment.erase(pos, link.size() + 1); // Erase trailing white space too.
-                string::size_type endpos = comment.find('}', pos);
-                if (endpos != string::npos)
-                {
-                    string ident = comment.substr(pos, endpos - pos);
-                    comment.erase(pos, endpos - pos + 1);
-
-                    //
-                    // Check for links of the form {@link Type#member}.
-                    //
-                    string::size_type hash = ident.find('#');
-                    string rest;
-                    if (hash != string::npos)
-                    {
-                        rest = ident.substr(hash + 1);
-                        ident = ident.substr(0, hash);
-                        if (!ident.empty())
-                        {
-                            ident = fixIdent(ident);
-                            if (!rest.empty())
-                            {
-                                ident += "." + fixIdent(rest);
-                            }
-                        }
-                        else if (!rest.empty())
-                        {
-                            ident = fixIdent(rest);
-                        }
-                    }
-                    else
-                    {
-                        ident = fixIdent(ident);
-                    }
-
-                    comment.insert(pos, ident);
-                }
-            }
-        } while (pos != string::npos);
-
-        StringList result;
-
-        pos = 0;
-        string::size_type nextPos;
-        while ((nextPos = comment.find_first_of('\n', pos)) != string::npos)
-        {
-            result.push_back(IceInternal::trim(string(comment, pos, nextPos - pos)));
-            pos = nextPos + 1;
-        }
-        string lastLine = IceInternal::trim(string(comment, pos));
-        if (!lastLine.empty())
-        {
-            result.push_back(lastLine);
-        }
-
-        trimLines(result);
-
-        return result;
-    }
-
-    bool parseCommentLine(const string& l, const string& tag, bool namedTag, string& name, string& doc)
-    {
-        doc.clear();
-
-        if (l.find(tag) == 0)
+        string::size_type hashPos = identifier.find("#");
+        if (hashPos != string::npos)
```

Comment:
This link formatter seems much complicated than others, and the fixIdent here and in others doesn't look totally correct.

I think would be simpler if we have a method to convert a Slice link identifier into a list of Slice identifiers, that can be used by all formatters.

Then the formatter can use the language mapping method for fixing keyworkds, and put back a link from the list.

Reply:
I reworked the parser to pre-split the link if a `#` is present.